librz/analysis/type_pdb.c

```diff
@@ -28,237 +49,634 @@ static char *create_type_name_from_offset(ut64 offset) {
+		typ->callable->ret = parse_type(typedb, ret_type);
+		if (!typ->callable->ret) {
+			typ->callable->noret = true;
+		}
+	}
+	// parse parameter list
+	SType *arglist = get_stype_by_index(lf_procedure->arg_list);
+	if (arglist) {
+		typ->callable->args = parse_type_arglist(typedb, &arglist->type_data);
+	}
+	return typ;
+}
+
+static RzType *parse_type_mfunction(const RzTypeDB *typedb, STypeInfo *type_info, char *name) {
+	SLF_MFUNCTION *lf_mfunction = type_info->type_info;
+	RzType *type = RZ_NEW0(RzType);
+	RzCallable *callable = RZ_NEW0(RzCallable);
+	if (!type || !callable) {
+		return NULL;
+	}
+	type->kind = RZ_TYPE_KIND_CALLABLE;
+	type->callable = callable;
+	type->callable->name = strdup(name);
+	type->callable->cc = parse_calling_convention(lf_mfunction->call_conv);
+	// parse return type
+	SType *ret_type = get_stype_by_index(lf_mfunction->return_type);
+	if (ret_type) {
+		type->callable->ret = parse_type(typedb, ret_type);
+		if (!type->callable->ret) {
+			type->callable->noret = true;
+		}
+	}
+	// parse parameter list
+	SType *arglist = get_stype_by_index(lf_mfunction->arglist);
+	if (arglist) {
+		type->callable->args = parse_type_arglist(typedb, &arglist->type_data);
+	}
+	return type;
+}
+
+static RzType *parse_type_onemethod(const RzTypeDB *typedb, STypeInfo *type_info) {
+	SLF_ONEMETHOD *lf_onemethod = type_info->type_info;
+	char *name = type_info->get_name(type_info);
+	SType *utype = get_stype_by_index(lf_onemethod->index);
+	if (!utype) {
+		return NULL;
+	}
+	STypeInfo *utype_info = &utype->type_data;
+	if (utype_info->leaf_type == eLF_MFUNCTION) {
+		return parse_type_mfunction(typedb, &utype->type_data, name);
+	}
+	return NULL;
+}
+
+static RzType *parse_type_member(const RzTypeDB *typedb, STypeInfo *type_info) {
+	SLF_MEMBER *lf_member = type_info->type_info;
+	SType *utype = get_stype_by_index(lf_member->index);
+	if (!utype) {
+		return NULL;
+	}
+	return parse_type(typedb, utype);
+}
+
+static RzType *parse_type_nest(const RzTypeDB *typedb, STypeInfo *type_info) {
+	SLF_NESTTYPE *lf_nest = type_info->type_info;
+	SType *utype = get_stype_by_index(lf_nest->index);
+	if (!utype) {
+		return NULL;
+	}
+	if (utype->type_data.get_name) {
+		char *name = utype->type_data.get_name(utype);
+		if (name) {
+			RzBaseType *b_type = rz_type_db_get_base_type(typedb, name);
+			if (b_type && b_type->type) {
+				if (b_type->type->kind == RZ_TYPE_KIND_IDENTIFIER) {
+					return rz_type_clone(b_type->type);
+				}
+			}
+			return NULL;
+		}
+	}
+
+	RzType *n_type = parse_type(typedb, utype);
+	if (!n_type) {
+		return NULL;
+	}
+	RzBaseType *btype = rz_type_base_type_new(RZ_BASE_TYPE_KIND_TYPEDEF);
+	if (!btype) {
+		return NULL;
+	}
+	btype->name = create_type_name_from_offset(lf_nest->index);
+	btype->type = n_type;
+	rz_type_db_save_base_type(typedb, btype);
+	if (n_type->kind == RZ_TYPE_KIND_IDENTIFIER) {
+		return rz_type_clone(n_type);
+	}
+	return NULL;
+}
+
 /**
- * @brief Parses class/struct/union member
```

Comment:
Please port to `\brief` from `@brief` doxygen syntax: https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#documentation